### PR TITLE
feat(frontend): sign up bb-9

### PR DESCRIPTION
**Pull request by iqbalalayubbi**
QA Notes:

- [x] Sign up action flow - user at this point is not relocated to the next page after clicking on Create an account. User should be relocated to the next page which is Onboarding questions according to the Figma (for now, i redirect to root)
- [x] Instead "Sign Up" there should be "CREATE AN ACCOUNT" according to Figma
- [x] The space between "Already have an account ..." and Name field is smaller than in Figma. Please adjust
- [x] The font size of the CREATE AN ACCOUNT text should be 24 according to the Figma. Now it's 28
- [x] There are no white dots in the right side of the screen as it is in design
- [x] In Sign in text "In" should start from upper case

Screenshot from 2024-08-27 22-40-15

### CODE REVIEW

File: apps/frontend/CHANGELOG.md
Comment (fshabanov) [inline apps/frontend/CHANGELOG.md:1]: You shouldn't have changes in this file

File: apps/frontend/src/pages/not-found/not-found.tsx [hunk 1]
```diff
@@ -9,9 +9,7 @@ const NotFound: React.FC = () => (
 		<p className={styles["text"]}>
 			The page you are looking for does not exist.
 		</p>
-		<Link to={AppRoute.ROOT} type="button">
-			Go to Homepage
-		</Link>
+		<Link to={AppRoute.ROOT}>Go to Homepage</Link>
```
Comment (fshabanov) [inline apps/frontend/src/pages/not-found/not-found.tsx:12]: Why was this changed?

**Review by fshabanov — CHANGES_REQUESTED** [review]
The point about redirecting the user does not seem to be fixed.

File: apps/frontend/src/pages/not-found/not-found.tsx (hunk 1 above)
Comment (iqbalalayubbi) [reply apps/frontend/src/pages/not-found/not-found.tsx:12]: my bad, it's maybe come when i try to merge with main branch

File: apps/frontend/CHANGELOG.md
Comment (iqbalalayubbi) [reply apps/frontend/CHANGELOG.md:1]: Sorry, i'm forget to check this file again after merge with main branch

File: apps/frontend/src/pages/auth/libs/components/sign-up-form/sign-up-form.tsx [hunk 2]
```diff
@@ -24,13 +24,15 @@ const SignUpForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
...
 
+	const navigate = useNavigate();
 	const handleFormSubmit = useCallback(
 		(event_: React.BaseSyntheticEvent): void => {
 			void handleSubmit((payload: UserSignUpFormDto) => {
 				const { confirmPassword, ...userData } = payload;
 
 				if (confirmPassword === userData.password) {
 					onSubmit(userData);
+					navigate("/");
```
Comment (JD-Rosales) [inline apps/frontend/src/pages/auth/libs/components/sign-up-form/sign-up-form.tsx:35]: I think it's better if we use the useSelector hook to check if the user is authenticated.
`if (user) {
			navigate(AppRoute.ROOT);
}`

**Review by JD-Rosales — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/auth/libs/components/sign-up-form/sign-up-form.tsx [hunk 3]
```diff
@@ -31,6 +47,7 @@ const SignUpForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
 
 				if (confirmPassword === userData.password) {
 					onSubmit(userData);
+					navigate(AppRoute.ROOT);
```
Comment (fshabanov) [inline apps/frontend/src/pages/auth/libs/components/sign-up-form/sign-up-form.tsx:50]: What if Backend throws an error?
The redirect logic should be handled in slice

File: apps/frontend/src/pages/auth/libs/components/sign-up-form/sign-up-form.tsx [hunk 4]
```diff
@@ -15,6 +22,15 @@ type Properties = {
 };
 
 const SignUpForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
+	const user = useAppSelector(({ auth }) => auth.user);
+	const navigate = useNavigate();
+
+	useEffect(() => {
+		if (user) {
+			navigate(AppRoute.ROOT);
```
Comment (fshabanov) [inline apps/frontend/src/pages/auth/libs/components/sign-up-form/sign-up-form.tsx:30]: User `<Navigate />` component in auth.tsx file

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Comment by iqbalalayubbi** [comment]
For redirected to root path already implemented with @JD-Rosales thank you

**Comment by iqbalalayubbi** [comment]
for the error handling can i implement like this?

error-handling

File: package-lock.json
Comment (fshabanov) [inline package-lock.json:1]: Shouldn't change this file

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (iqbalalayubbi) [reply package-lock.json:1]: but when i run npm install, this file change because there are some package install

File: apps/frontend/src/modules/auth/slices/actions.ts [hunk 5]
```diff
@@ -28,15 +28,24 @@ const signUp = createAsyncThunk<
...
+			const { token, user } = await authApi.signUp(registerPayload);
 
-	void storage.set(StorageKey.TOKEN, token);
+			void storage.set(StorageKey.TOKEN, token);
 
-	return user;
-});
+			return user;
+		} catch (error) {
+			const errorMessage = (error as Error).message;
```
Comment (ihorLysak) [inline apps/frontend/src/modules/auth/slices/actions.ts:43]: `throw error as Error` should be sufficient

File: apps/frontend/src/modules/auth/slices/actions.ts [hunk 6]
```diff
@@ -28,15 +28,24 @@ const signUp = createAsyncThunk<
...
+		try {
+			const { token, user } = await authApi.signUp(registerPayload);
 
-	void storage.set(StorageKey.TOKEN, token);
+			void storage.set(StorageKey.TOKEN, token);
 
-	return user;
-});
+			return user;
+		} catch (error) {
```
Comment (fshabanov) [inline apps/frontend/src/modules/auth/slices/actions.ts:42]: We don't need try catch block here, all errors will be handled by middleware

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Comment by fshabanov** [comment]
Merge the main branch

**Review by WilhelmWesser — APPROVED** [review]
